byterun/obj.c

```diff
@@ -150,13 +150,34 @@ CAMLprim value caml_obj_compare_and_swap (value v, value f, value oldv, value ne
   return Val_int(res);
 }
 
-/* caml_promote_to(obj, upto) promotes obj to be as least as shared as upto */
+/* caml_promote_to(obj, upto) may promote obj to be as least as shared as upto.
+   Skips promotion if there is only a single domain. This should only be used in
+   situations where there is a performance need to 'front-load' the promotion
+   across domains, such as for queues. If you need to guarantee promotion to the
+   major heap for correctness, */
 CAMLprim value caml_obj_promote_to (value obj, value upto)
 {
-  if (Is_block(upto) && Is_minor(upto)) {
+  if (Is_block(upto) && (Is_minor(upto) || caml_domain_alone()))
+  {
```

Comment:
Indentation in the runtime is [K&R style](https://en.wikipedia.org/wiki/Indentation_style#K&R_style): function opening braces go on their own line, but other opening braces (for `if`, `while`, etc.) go on the same line as their statement.

Reply:
Thanks. This should be resolved now.